src/api/java/com/minecolonies/api/tileentities/TileEntityRack.java

```diff
@@ -225,12 +225,14 @@ public void updateItemStorage()
                 continue;
             }
 
-            final ItemStorage storage = new ItemStorage(stack.copy());
+            final ItemStack stackCopy = stack.copy();
+            final ItemStorage storage = new ItemStorage(stackCopy);
             int amount = ItemStackUtils.getSize(stack);
             if (content.containsKey(storage))
             {
                 amount += content.remove(storage);
             }
+            stackCopy.setCount(amount);
             content.put(storage, amount);
         }
```

Comment:
```
int amount = ItemStackUtils.getSize(stack);
if (content.containsKey(storage))
{
amount += content.remove(storage);
}
content.put(storage, amount);
```
Why isn't this enough?

Reply:
because [this](https://github.com/ldtteam/minecolonies/blob/84fa0aaf06052529fb9c664f27a6f715c3b2bb2b/src/main/java/com/minecolonies/coremod/client/gui/WindowHutAllInventory.java#L150) only uses the stack, not the amount.
both in the current, and my changed version.

Reply:
that should work normally

Reply:
it gets the amount from the stack and adds it to the amount in the map. That should be enough

Reply:
Should i change it back to how it was in the second commit then? that didn't change TileEntityRack at all.

Reply:
none of this should be necessary of the above.

Reply:
does that mean you prefer leaving this alone like it was, and just changing the WindowHutAllInventory?

Reply:
The above code is correct. It should not be the cause of any issue.

Reply:
ok i changed it back.